### PR TITLE
winpr/input: Add actual evdev keyboard handling

### PR DESCRIPTION
**Pull request by pnowack**
Quoting the commit message here:

```
WinPR provides APIs to convert between keycodes between virtual
keycodes.
These keycodes can currently be evdev keycodes or Apple keycodes.
The evdev handling, however, handles XKB keycodes and not evdev ones.
The main difference between these is that XKB keycodes are shifted by
the value 8, compared to evdev keycodes.

In order to fix this situation, rename the evdev keycodes to XKB ones,
and introduce additionally a new keycode evdev, including its handling
for this keycode type.
```

This is an API breaking change.

### CODE REVIEW

**Comment by pnowack** [comment]
Did split up a few lines to have both a consistent indentation for the comments and to make clang-format happy.

File: winpr/libwinpr/input/keycode.c [hunk 1]
```diff
@@ -623,6 +902,17 @@ DWORD GetKeycodeFromVirtualKeyCode(DWORD vkcode, DWORD dwFlags)
 			}
 		}
 	}
+	else if (dwFlags & KEYCODE_TYPE_XKB)
+	{
+		for (index = 0; index < 256; index++)
```
Comment (akallabeth) [inline winpr/libwinpr/input/keycode.c:907]: Maybe use `ARRAYSIZE(KEYCODE_TO_VKCODE_XKB)` here?

Comment (pnowack) [reply winpr/libwinpr/input/keycode.c]: Yep, did also the same for `KEYCODE_TO_VKCODE_APPLE`, and `KEYCODE_TO_VKCODE_EVDEV`.

**Comment by pnowack** [comment]
@freerdp-bot test

**Comment by hardening** [comment]
Suggestion: if that breaks API/API we shall start a file to document all these changes, and give some hints of how to migrate to 3.0

**Comment by pnowack** [comment]
> Suggestion: if that breaks API/API we shall start a file to document all these changes, and give some hints of how to migrate to 3.0

Indeed. Do we already have something like an empty wiki page, where markdown works (something which we members can all edit)?

**Review by akallabeth — APPROVED** [review]
LGTM

**Pull request merged**